NitroxServer/GameLogic/Entities/EntitySimulation.cs

```diff
@@ -111,14 +111,17 @@ public SimulatedEntity AssignNewEntityToPlayer(Entity entity, Player player, boo
         throw new Exception($"New entity was already being simulated by someone else: {entity.Id}");
     }
 
-    public IEnumerable<NitroxId> AssignGlobalRootEntities(Player player)
+    public List<SimulatedEntity> AssignGlobalRootEntitiesAndGetData(Player player)
     {
-        IEnumerable<WorldEntity> entities = worldEntityManager.GetGlobalRootEntities()
-                                                              .Where(entity => simulationOwnershipData.TryToAcquire(entity.Id, player, SimulationLockType.TRANSIENT));
-        foreach (WorldEntity entity in entities)
+        List<SimulatedEntity> simulatedEntities = new();
+        foreach (GlobalRootEntity entity in worldEntityManager.GetGlobalRootEntities())
         {
-            yield return entity.Id;
+            simulationOwnershipData.TryToAcquire(entity.Id, player, SimulationLockType.TRANSIENT);
```

Comment:
Yes, this is just an attempt to acquire the lock without caring about how it ends up because on the 2nd line after this, we get the player which actually owns the lock.